### PR TITLE
Feature 2: User checks off task from ToDoList, added test files and enum object "Priority"

### PR DESCRIPTION
**Pull request by XinyingXu**
Updated use cases CompleteTaskUC.java and GainPointsUC.java to meet the naming standards; updated methods regarding checklists, but still need a searchFor(String taskName) method to locate a task in the ToDoList.

Added test file for User.java.

Added an enum object Priority, which acts as an attribute of the Task object.

### CODE REVIEW

File: src/main/java/controllers/CompleteTaskControl.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
+package controllers;
+
+import useCases.CompleteTaskUC;
+import useCases.GainPointsUC;
+
+public class CompleteTaskControl {
+    //the UI passed the Task name to controller
+    String taskName;
```
Comment (e-laflamme) [inline src/main/java/controllers/CompleteTaskControl.java:8]: For a future implementation we may want to figure out how to find the task in a way that does not require each to have a unique name.

File: src/main/java/useCases/CompleteTaskUC.java [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+
+import entities.Priority;
+import presenters.CompleteTaskResponseModel;
+
+public class CompleteTaskUC {
+    private Task task;
+    /* checklists acquired from DataAccess */
+    private ToDolist todo;
```
Comment (e-laflamme) [inline src/main/java/useCases/CompleteTaskUC.java:11]: Should add a method to get these variables.

File: src/main/java/useCases/GainPointsUC.java [hunk 3]
```diff
@@ -0,0 +1,37 @@
+package useCases;
+import entities.User;
+import entities.Priority;
+import presenters.CompleteTaskResponseModel;
+import presenters.CompleteTaskPresenter;
+
+public class GainPointsUC {
+    CompleteTaskUC ct;
+    User user;
```
Comment (e-laflamme) [inline src/main/java/useCases/GainPointsUC.java:9]: Need to add a method to get the User. We should discuss as a group how this should be done.

**Review by e-laflamme — APPROVED** [review]
Awesome! For next steps after merge I have made a few comments on things that could be worked on.

**Review by serenahonekin — APPROVED** [review]
There may not be a need for the completeTaskReponse model depending on the data access we are using

**Comment by XinyingXu** [comment]
> There may not be a need for the completeTaskReponse model depending on the data access we are using

Yeah maybe. For now we can just include these because of clean architecture, I can delete them if they are no longer needed in the future.

File: src/main/java/useCases/CompleteTaskUC.java [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+    //Check if the task is finished before studyDeadline
+    public Boolean finishedBeforeDDL () {
+        LocalDateTime now = LocalDateTime.now();
+        deadline = task.getStudyDeadline();//StudyDeadline attribute of Task, which is a date/time
+        return deadline.compareTo(now);
+    }
+
+    //return the task priority
+    public Priority getPriority () {
+        return task.getPriority();
```
Comment (sing1348) [inline src/main/java/useCases/CompleteTaskUC.java:29]: maybe add a feature to be able to view tasks based on a chosen priority

**Pull request merged**